tools/qc.wdl

```diff
@@ -4,57 +4,40 @@
 
 version 1.0
 
-task parse_infer_experiment {
+task parse_validate_bam {
     input {
         String in
+        Boolean strict = true
         Int max_retries = 1
     }
-
-    command { 
-        if [ "$(echo "~{in}" | grep -c 'PairEnd')" -gt 0 ]
-        then
-            if (( $(echo "$(echo "~{in}" | tail -n 1 | sed  's/^.*: //') > 0.5" | bc -l ) ))
-            then
-               echo 'strand-specific-reverse' > stdout.txt
-            elif (( $(echo "$(echo "~{in}" | tail -n 2 | head -n 1 | sed  's/^.*: //') > 0.5" | bc -l ) ))
-            then 
-               echo 'strand-specific-forward' > stdout.txt
-            else
-               echo 'non-strand-specific' > stdout.txt
-            fi
-        elif [ "$(echo "~{in}" | grep -c 'SingleEnd')" -gt 0 ]
+ 
+    command {
+        if [ "~{strict}" == "true" ]
         then 
-            if (( $(echo "$(echo "~{in}" | tail -n 1 | sed  's/^.*: //') > 0.5" | bc -l ) ))
-            then
-               echo 'strand-specific-reverse' > stdout.txt
-            elif (( $(echo "$(echo "~{in}" | tail -n 2 | head -n 1 | sed  's/^.*: //') > 0.5" | bc -l ) ))
-            then 
-               echo 'strand-specific-forward' > stdout.txt
-            else
-               echo 'non-strand-specific' > stdout.txt
-            fi
-        else
-           echo "infer_experiment failed to determine type" > /dev/stderr
-           exit 1 
+           if [ "$(echo "~{in}" | grep -c "ERROR")" -gt 0 ]
```

Comment:
I think this can be cleaner. Just set the string to grep.

```
if [ "~{strict}" == "true" ]; then
  GREP_PATTERN="ERROR"
else
  GREP_PATTERN="(ERROR|WARNING)"
fi

if [ "$(echo "~{in}" | grep -Ec "$GREP_PATTERN")" -gt 0 ]
```

Reply:
Also, does that last line actually work with double quotes embedded in double quotes?

Reply:
@claymcleod @jsunny23 Was going to make that suggested change, realized essentially every piece of validate sam was broken. However the double quotes is fine, which is what I was expecting to be the problem. Basically the output was never being saved correctly, but should be good now.